python/ray/tests/test_ray_shutdown.py

```diff
@@ -51,6 +51,61 @@ def f():
     wait_for_condition(lambda: len(get_all_ray_worker_processes()) == 0)
 
 
+@pytest.mark.skipif(platform.system() == "Windows", reason="Hang on Windows.")
+def test_ray_shutdown_then_call(short_gcs_publish_timeout, shutdown_only):
+    """Make sure ray will not kill cpython when using unrecognized ObjectId"""
+    # Set include_dashboard=False to have faster startup.
+    ray.init(num_cpus=1, include_dashboard=False)
+
+    my_ref = ray.put("anystring")
+
+    @ray.remote
+    def f(s):
+        print(s)
+
+    ray.shutdown()
+
+    ray.init(num_cpus=1, include_dashboard=False)
+    exception_thrown = False
+    try:
+        f.remote(my_ref)  # This would cause full CPython death.
+    except ray.exceptions.RayError:
```

Comment:
Could we check more specifically to make sure the exact error is thrown, RayError is too broad. We can even check the exception message.

Btw, you can use `with pytest.raises(...)`

Reply:
done. thanks!